### ISSUE TITLE
[Regression] error: undefined symbol: __cxa_begin_catch

### ISSUE THREAD

**Issue by kkimdev** [emscripten-core/emscripten#8856]
Minimal reproducible steps:
```
touch main.cpp
em++ main.cpp -fno-exceptions -s DEMANGLE_SUPPORT=1
```

output:
```
error: undefined symbol: __cxa_begin_catch
warning: To disable errors for undefined symbols use `-s ERROR_ON_UNDEFINED_SYMBOLS=0`
Error: Aborting compilation due to previous errors
shared:ERROR: '/opt/emsdk/node/8.9.1_64bit/bin/node /opt/emsdk/fastcomp/emscripten/src/compiler.js /tmp/tmpxV5xUR.txt /opt/emsdk/fastcomp/emscripten/src/library_pthread_stub.js' failed (1)
```

Tested version: 1.38.36

The same codebase(not the above minimized version) worked before, and stopped working once I updated to 1.38.36

**Comment by kripken** [comment]
Thanks @kkimdev, I can confirm this is a regression. It happened in https://github.com/emscripten-core/emscripten/pull/8040 - @juj, that's your PR, can you please take a look?

**Comment by sbc100** [comment]
I'm not sure if its related but there is a hack in tools/system_libs.py that tried to put `libc++abi.bc` last on the link line but it doesn't do it for `libc++abi.a`.

**Comment by VirtualTim** [comment]
@kkimdev In case you didn't figure it out already, you should be able to work aound this by compiling with something like `-s EXPORTED_FUNCTIONS=['__cxa_begin_catch']`.
Not a permanent solution, but should get you back up and running.

**Comment by juj** [comment]
Hmm.. does this still occur for you? Testing on current `incoming` hash 2aeae940c039e16cfeb19404c38e89a34a5cb514 the STR does not reproduce the issue for me.

**Comment by kkimdev** [comment]
I haven't tested but if it's not reproducible on the incoming branch I guess it's fixed.

**Comment by nickshinpho** [comment]
this error using Minimal reproducible steps is happening in 1.38.45 [a400afe](https://github.com/emscripten-core/emscripten/commit/a400afe9c0aeb89fc800a344bd8765e44119ea35)

i even tried using VirtualTim's suggestion, and am still getting the undefined symbol error.

any ideas and tips on how to fix this?

thanks in advanced!

**Comment by kripken** [comment]
Thanks @nickshinpho, looks like this was fixed in upstream but not in fastcomp (dce differences, probably), and we missed that.

A fix + testing is in #9478 which I verified fixes the testcase here in both backends.
